pkg/executor/test/admintest/admin_test.go

```diff
@@ -1773,3 +1779,219 @@ func TestAdminCheckTableErrorLocateForClusterIndex(t *testing.T) {
+		require.NoError(t, err)
+		txn, err = store.Begin()
+		require.NoError(t, err)
+		txn.Delete(tablecodec.EncodeIndexSeekKey(tblInfo.ID, idxInfo.ID, idxVal))
+		err = txn.Commit(context.Background())
+		require.NoError(t, err)
+		err = tk.ExecToErr("admin check table admin_test")
+		require.Error(t, err)
+		require.True(t, consistency.ErrAdminCheckInconsistent.Equal(err))
+		require.EqualError(t, err, "[admin:8223]data inconsistency in table: admin_test, index: uidx_a, handle: 2, index-values:\"\" != record-values:\"handle: 2, values: [KindInt64 -1]\"")
+
+		// Add one row with inconsistent value.
+		// Index count = table count, but data is different.
+		rd := rowcodec.Encoder{Enable: true}
+		value, err := tablecodec.EncodeRow(tk.Session().GetSessionVars().StmtCtx.TimeZone(),
+			[]types.Datum{types.NewIntDatum(100), types.NewIntDatum(200), types.NewIntDatum(300)},
+			[]int64{tblInfo.Cols()[0].ID, tblInfo.Cols()[1].ID, tblInfo.Cols()[2].ID}, nil, nil, &rd)
+		require.NoError(t, err)
+		txn, err = store.Begin()
+		require.NoError(t, err)
+		err = txn.Set(tablecodec.EncodeRowKeyWithHandle(df.ID, kv.IntHandle(2)), value)
+		require.NoError(t, err)
+		err = txn.Commit(context.Background())
+		require.NoError(t, err)
+		err = tk.ExecToErr("admin check table admin_test")
+		require.Error(t, err)
+		require.True(t, consistency.ErrAdminCheckInconsistent.Equal(err))
+		require.ErrorContains(t, err, "[admin:8223]data inconsistency in table: admin_test, index: uidx_a, ")
+	}
+}
+
+func TestAdminCheckGlobalIndexDuringDDL(t *testing.T) {
+	store, dom := testkit.CreateMockStoreAndDomain(t)
+	originalHook := dom.DDL().GetHook()
+	tk := testkit.NewTestKit(t, store)
+
+	var schemaMap = make(map[model.SchemaState]struct{})
+
+	hook := &callback.TestDDLCallback{Do: dom}
+	onJobUpdatedExportedFunc := func(job *model.Job) {
+		schemaMap[job.SchemaState] = struct{}{}
+		_, err := tk.Exec("admin check table admin_test")
+		assert.NoError(t, err)
+	}
+	hook.OnJobUpdatedExported.Store(&onJobUpdatedExportedFunc)
+
+	// check table after delete some index key/value pairs.
+	ddl.MockDMLExecution = func() {
+		_, err := tk.Exec("admin check table admin_test")
+		assert.NoError(t, err)
+	}
+
+	var enableFastCheck = []bool{false, true}
+	for _, enabled := range enableFastCheck {
+		tk.MustExec("use test")
+		tk.MustExec("drop table if exists admin_test")
+
+		tk.MustExec("set tidb_enable_global_index = true")
+		tk.MustExec(fmt.Sprintf("set tidb_enable_fast_table_check = %v", enabled))
+
+		tk.MustExec("create table admin_test (a int, b int, c int, unique key uidx_a(a), primary key(c)) partition by hash(c) partitions 5")
+		tk.MustExec("insert admin_test values (-10, -20, 1), (-1, -10, 2), (1, 11, 3), (2, 12, 0), (5, 15, -1), (10, 20, -2), (20, 30, -3)")
+
+		dom.DDL().SetHook(hook)
+		require.NoError(t, failpoint.Enable("github.com/pingcap/tidb/pkg/ddl/mockDMLExecution", "1*return(true)->return(false)"))
+		tk.MustExec("alter table admin_test truncate partition p0")
```

Comment:
It seems the p0 only has 1 record(2, 12, 0), so it feels like it's the same as doing the `admin check` right after the truncate operation.

Reply:
Done

Reply:
There's no difference between P0 (1 record) and p1 (2 records), because there seems to be a batch process in the backfill?